File: shared/src/main/java/xyz/klinker/messenger/shared/util/listener/MediaSelectedListener.java
package xyz.klinker.messenger.shared.util.listener;

import java.util.List;

import xyz.klinker.messenger.shared.data.model.Message;

/**
 * Callback for easily notifying the caller when a media has been selected
 */
public interface MediaSelectedListener {
    void onSelected(List<Message> messageList, int selectedPosition);
}

File: shared/src/main/java/xyz/klinker/messenger/shared/util/listener/MediaSelectedListener.kt
package xyz.klinker.messenger.shared.util.listener

import xyz.klinker.messenger.shared.data.MediaMessage
import xyz.klinker.messenger.shared.data.model.Message

/**
 * Callback for easily notifying the caller when a media has been selected
 */
interface MediaSelectedListener {
    fun onSelected(messageList: List<MediaMessage>, selectedPosition: Int)
    fun onStartDrag(index: Int)
}
